.github/workflows/update_readme.yml

```diff
@@ -0,0 +1,36 @@
+# -----------------------------------------------------------------------------
+#   - invoked on PR to main branch
+# -----------------------------------------------------------------------------
+name: update README
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  update-readme:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.11
+      - name: Update README
+        run: |
+          pip install packaging
+          python update_readme.py
+      - name: Commit and push changes
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "github-actions"
+          git diff --quiet && git diff --staged --quiet || (
```

Comment:
I don't think it's right for this to commit directly to main, since it's triggered by a pull request targeting main. It might make sense if it was triggered by a commit to main, but since it's triggered by a pull request, it should commit to the branch of the pull request imo.

Reply:
Ok, I'll need to work on this but basically we want to have this happen on any PUSH (b/c any given push could be modifying the python_versions.json)